File: C-Plus-Plus/cp/Reverse_Words_in_String.cpp
/**
 * @author omkarlanghe
 * Given a String of length S, reverse the whole string without reversing the individual words in it.
 * Words are separated by dots.
 * 
 * Input:
 * The first line contains T denoting the number of testcases. 
 * T testcases follow. Each case contains a string S containing characters.
 * 
 * Output:
 * For each test case, in a new line, output a single line containing the reversed String.
 * 
 * Example:
 * Input:
 * 2
 * i.like.this.program.very.much
 * pqr.mno
 * 
 * Output:
 * much.very.program.this.like.i
 * mno.pqr
*/
#include <iostream>
#include <stack>

int main() {
    int t;
    std::cout << "Enter test cases : " << std::endl;
    std::cin >> t;
    while (t--) {
        std::stack<std::string> str_stack;
        std::string str, stack_string;

        std::cout << "Enter the string : " << std::endl;
        std::cin >> str;

        for (int i = 0 ; i < str.size() ; i++) {
            if (str[i] != '.') {
                stack_string += str[i];
            } else {
                str_stack.push(stack_string);
                str_stack.push(".");
                stack_string = "";
            }
        }
        str_stack.push(stack_string);
        str = "";
        while (!str_stack.empty()) {
            str += str_stack.top();
            str_stack.pop();
        }
        std::cout << str << std::endl;
    }
    return (0);
}

File: Java/cp/Reverse_words_in_String.java
    /*
    Given a string, we need to reverse it without reversing any word in it.
    We can reverse each word of a string.
    In other words, Reverse text without reversing individual word in a String.
    */
    import java.util.Scanner;

    public class Reverse_String 
    {
    /*
    This is an constructor called from main function
    */
    public Reverse_String(String in[],int n)
    {
         System.out.println(); 
        
	   for (int i = n - 1; i >= 0; i--) 
           {
               System.out.print(in[i] + " ");
           }
    }
    public static void main(String[] args) 
    {

       Scanner scan = new Scanner(System.in);
       System.out.println("Please Enter the number of words your String have....");
       int n = scan.nextInt();
       String in[] = new String[n];
       System.out.println("Please enter your String here...");
           for (int i = 0; i <= n-1; i++) 
	   {
              in[i] = scan.next();
           }
       Reverse_String r1=new Reverse_String(in,n);
    }
    }   
    /*
    Time complexity is O(n-1)
    Space complexity is O(1)
    Input :
    Please Enter the number of words your String have....
    6
    Please enter your String here...
    i like this program very much
    Output:
    much very program this like i
    */
